test/integration/thermal_system.cc

```diff
@@ -76,20 +79,49 @@ TEST_F(ThermalTest, TemperatureComponent)
             // verify temperature data belongs to a visual
             EXPECT_NE(nullptr, _ecm.Component<components::Visual>(_id));
 
+            return true;
+          });
+
+      _ecm.Each<components::TemperatureRange, components::SourceFilePath,
+        components::Name>(
+          [&](const ignition::gazebo::Entity &_id,
+              const components::TemperatureRange *_tempRange,
+              const components::SourceFilePath *_heatSigURI,
+              const components::Name *_name) -> bool
+          {
+            // store temperature range data
+            entityTempRange[_name->Data()] = _tempRange->Data();
+
+            // store heat signature URI data
+            heatSignatures[_name->Data()] = _heatSigURI->Data();
+
+            // verify temperature range data belongs to a visual
+            EXPECT_NE(nullptr, _ecm.Component<components::Visual>(_id));
+
             return true;
           });
     });
   server.AddSystem(testSystem.systemPtr);
 
-  // verify nothing in map at beginning
+  // verify nothing in the maps at beginning
   EXPECT_TRUE(entityTemp.empty());
+  EXPECT_TRUE(entityTempRange.empty());
+  EXPECT_TRUE(heatSignatures.empty());
 
   // Run server
   server.Run(true, 1, false);
 
   // verify temperature components are created and the values are correct
-  EXPECT_EQ(3u, entityTemp.size());
-  EXPECT_DOUBLE_EQ(200.0, entityTemp["box_visual"].Kelvin());
+  EXPECT_EQ(2u, entityTemp.size());
+  EXPECT_EQ(2u, entityTempRange.size());
+  EXPECT_EQ(2u, heatSignatures.size());
```

Comment:
After these checks, it would probably be worth adding some `ASSERT_TRUE` checks to make sure that the keys we are about to check below are actually in the maps (if they're not in the maps, this could lead to undesirable behavior, so stop now)

Reply:
fixed in https://github.com/ignitionrobotics/ign-gazebo/pull/498/commits/30d5cf3da8f9190b1d338a52e043316ce2e00154